### PR TITLE
[Strings1] Improve string repr of PauliX, PauliY, PauliZ 

### PR DESCRIPTION
**Pull request by Qottmann**
Improving string representation of PauliX, PauliY and PauliZ to be less verbose.
Together with adding the alias for X, Y, Z and I, this makes the string outputs copy-and-paste-able.

```
>>> qml.PauliX(0)
X(0)
>>> qml.PauliX('a')
X('a')
```

Also tackling scalar products s.t.
```
>>> 0.5 * X(0) # no unnecessary bracket
0.5 * X(0)
>>> 0.5 * (X(0) + Y(1)) # keeps necessary bracket
0.5 * (X(0) + Y(1))
```
(all examples with new opmath `qml.operation.enable_new_opmath()`)

- [x] Update PauliX, PauliY, PauliZ, Identity string repr
- [x] Update tests accordingly
- [x] update SProd to not unnecessarily use brackets
- [x] add tests accordingly
- [x] Add alias for X, Y, Z and I
- [x] add tests accordingly
- [x] fix potential issues for classical shadows
- [x] changelog
- [x] handle Identity()

[sc-52831]
[sc-54957]
[sc-43897]

### CODE REVIEW

File: tests/ops/qubit/test_non_parametric_ops.py [hunk 1]
```diff
@@ -67,6 +67,24 @@
...
+    (qml.PauliZ(0), "Z(0)"),
+    (qml.Identity("a"), "I('a')"),
+    (qml.PauliX("a"), "X('a')"),
+    (qml.PauliY("a"), "Y('a')"),
+    (qml.PauliZ("a"), "Z('a')"),
+)
+
+
+@pytest.mark.parametrize("wire", [0, "a", "a"])
+def test_alias_XYZ(wire):
```
Comment (albi3ro) [inline tests/ops/qubit/test_non_parametric_ops.py:83]: Might be worth adding an Identity test here too.

**Review by albi3ro — COMMENTED** [review]
Looks so much cleaner 🎉 This will be a massive quality of life improvement for everyone!

We should potentially add a note about the syntax on the "Quantum operators" page:

https://docs.pennylane.ai/en/stable/introduction/operations.html#qubit-operators

Changelog entry, and a confirmed sign-off from @trbromley as this has a lot of user facing consequences, and I'll be happy to approve :)

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -165,6 +165,19 @@
   and the codecov check itself would never execute.
   [(#5101)](https://github.com/PennyLaneAI/pennylane/pull/5101)
 
+* String representations of Pauli operators have been improved and there are new aliases `X, Y, Z, I` for `PauliX, PauliY, PauliZ, Identity`.
+  ```
+  >>> qml.PauliX(0)
+  X(0)
+  >>> qml.PauliX('a')
+  X('a')
+  >>> 0.5 * X(0) # no unnecessary bracket
```
Comment (trbromley) [inline doc/releases/changelog-dev.md:174]: Minor point but the comments on the unnecessary bracket feel harder to understand if you don't know the "before" version.

**Review by trbromley — APPROVED** [review]
Thanks @Qottmann!

File: doc/releases/changelog-dev.md (hunk 2 above)
Comment (Qottmann) [reply doc/releases/changelog-dev.md:174]: So just not make that comment and say "here is how (nice) it looks now" 🙂 ?

**Review by albi3ro — APPROVED** [review]

**Pull request merged**